controllers/openstacknet_controller.go

```diff
@@ -197,11 +220,97 @@ func (r *OpenStackNetReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		err = common.CreateOrUpdateNetworkAttachmentDefinition(r, instance, instance.Kind, metav1.NewControllerRef(instance, instance.GroupVersionKind()), &nad)
 		if err != nil {
+			instance.Status.CurrentState = ospdirectorv1beta1.NetError
+			_ = r.setStatus(instance, oldStatus, err.Error())
 			return ctrl.Result{}, err
 		}
+
+		// If we get this far, we assume the NAD been successfully created (NAD does not
+		// have a status block we can examine), so now examine the status of the NNCP
+
+		//nncp := &nmstatev1alpha1.NodeNetworkConfigurationPolicy{}
+		nncpList := &nmstatev1alpha1.NodeNetworkConfigurationPolicyList{}
+
+		// Can't use "r.Client.Get" here, as the NNCP is cluster-scoped and does not have a namespace.  We require the
+		// NNCP resource struct object in order to examine its status, but the NNCP was created earlier via a server-side
+		// apply.  Thus we don't already have the struct object in memory, and must acquire it by asking for a list of
+		// NNCPs filtered by our label selector (which should return just the one resource we seek)
+		// err = r.Client.Get(context.TODO(), types.NamespacedName{Name: ncp.Name, Namespace: instance.Namespace}, nncp)
+		err = r.Client.List(context.TODO(), nncpList, &client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(common.GetLabels(instance, openstacknet.AppLabel, map[string]string{})),
+		})
+
+		if err != nil {
+			instance.Status.CurrentState = ospdirectorv1beta1.NetError
+			_ = r.setStatus(instance, oldStatus, err.Error())
+			return ctrl.Result{}, err
+		}
+
+		if len(nncpList.Items) < 1 {
+			msg := fmt.Sprintf("OpenStackNet %s associated NodeNetworkConfigurationPolicy is still initializing, requeuing...", instance.Name)
+			r.Log.Info(msg)
+			instance.Status.CurrentState = ospdirectorv1beta1.NetInitializing
+			err = r.setStatus(instance, oldStatus, msg)
+			return ctrl.Result{RequeueAfter: time.Second * 5}, err
+		}
+
+		nncp := nncpList.Items[0]
+
+		netState = ospdirectorv1beta1.NetConfiguring
+		netStateMsg = fmt.Sprintf("OpenStackNet %s is configuring targeted node(s)", instance.Name)
+
+		for _, condition := range nncp.Status.Conditions {
+			if condition.Status == corev1.ConditionTrue {
+				if condition.Type == nmstateapi.NodeNetworkConfigurationPolicyConditionAvailable {
+					netState = ospdirectorv1beta1.NetConfigured
+					netStateMsg = fmt.Sprintf("OpenStackNet %s has been successfully configured on targeted node(s)", instance.Name)
+					break
+				} else if condition.Type == nmstateapi.NodeNetworkConfigurationPolicyConditionDegraded {
+					instance.Status.CurrentState = ospdirectorv1beta1.NetError
+					_ = r.setStatus(instance, oldStatus, err.Error())
+					return ctrl.Result{}, err
+				}
+			}
+		}
 	}
 
-	return ctrl.Result{}, nil
+	// Count IP assignments and update status
+	totalIps := 0
+	activeIps := 0
+
+	for _, roleIps := range instance.Status.RoleReservations {
+		for _, roleIP := range roleIps.Reservations {
+			totalIps++
+			if !roleIP.Deleted {
```

Comment:
this could be confusing. the behavior right now is
- on a osvmset or osbmset delete, like role delete all IP reservations get returned to the pool
- on single system removal from an CR, the IP reservation is marked as deleted, but can not be consumed by another CR/role, so it is not completely deleted. if we have e.g. a role which used 50 IPs, then remove 40 nodes from that group, this will still have the 40 as deleted, but shown as only 10 be consumed, right?

Maybe we should just show the len of roleIps.Reservations and in a follow up discuss if we should return all deleted flagged IP reservations to be available for any role?

Reply:
We can certainly just show the sum of the length of all `roleIps.Reservations` for now and discuss further changes for later.  But yes, you're right.  In that scenario, if you had 50 and deleted 40, it would say 50 reserved and 10 active.